src/daemon/led_hw.c

```diff
@@ -20,45 +44,77 @@ int updatergb_wireless(usbdevice* kb, lighting* lastlight, lighting* newlight){
     return 0;
 }
 
-void cmd_hwanim(usbdevice* kb, usbmode* dummy1, int dummy2, int keyindex, const char* code){
-    // Format: animation mode:speed:random:colour:colour2
-    uchar anim = 0xff, speed = 2, random = 0;
-    uchar r[2] = {}, g[2] = {}, b[2] = {};
-    sscanf(code, "%hhx:%hhu:%hhu:%2hhx%2hhx%2hhx:%2hhx%2hhx%2hhx", &anim, &speed, &random, r, g, b, r+1, g+1, b+1);
-
-    short zone = kb->keymap[keyindex].led;
-    // Limit values
-    if(anim != HW_COLOURSHIFT && anim != HW_COLOURPULSE && anim != HW_RAINBOW && anim != HW_STATIC && anim != HW_NONE)
-        return;
-    if(speed < 1 || speed > 3)
-        return;
-    if(random != 0 && random != 1)
-        return;
-    if(zone < 0 || zone > 2)
-        return;
-
-    // Deal with the random bytes' weirdness
-    if((anim == HW_COLOURSHIFT || anim == HW_COLOURPULSE) && !random)
-        random = 3;
-
-    // Dark Core zone bits:
-    // 1: Front
-    // 2: Thumb
-    // 4: Rear
-
-    // Dark Core commands:
-    // 00: Colour Shift
-    // 01: Colour Pulse
-    // 03: Rainbow
-    // 07: Static Colour
-    // FF: No animation (black)
-
-    if(IS_DARK_CORE(kb)) {
-        uchar data_pkt[MSG_SIZE] = {
+void cmd_hwanim(usbdevice* kb, usbmode* dummy1, int dummy2, int keyindex, const char* code) {
+    /* 
+       Unified format for the code string:
+       - DARK CORE Format: animation mode:speed:random:colour:colour2
+       - K63 WL    Format: animation mode:speed:dir
+    */
+
+    if (IS_DARK_CORE(kb)) {
+        // DARK CORE Format: animation mode:speed:random:colour:colour2
+        uchar anim = 0xff, speed = 2, random = 0;
+        uchar r[2] ={}, g[2] ={}, b[2] ={};
+        sscanf(code, "%hhx:%hhu:%hhu:%2hhx%2hhx%2hhx:%2hhx%2hhx%2hhx", &anim, &speed, &random, r, g, b, r+1, g+1, b+1);
+
+        short zone = kb->keymap[keyindex].led;
+
+        // Limit values
+
+        switch (anim) {
```

Comment:
I would suggest keeping changes to a minimum here. I don't think we need the switches. Also, I think you've accidentally auto-formatted the `if()` to `if ()`, `for()` to `for ()`, etc. Please undo that.

Reply:
Oh, I wasn't aware of any coding style guidelines

Reply:
There really aren't any, I just kinda do what I find in the code.